kiwi/command.py

```diff
@@ -165,3 +173,24 @@ def _select():
             error_available=error_available(),
             process=process
         )
+
+    @classmethod
+    def _command_exists(self, command, env):
+        """
+        Tests if the given command is executable for the specified
+        environment. If the specified environment does not include
+        'PATH' variable, the method raises a KiwiCommandError
+        exception
+
+        :param string command: command to validate
+        :param list env: custom os.environ
+
+        :return: True if the command is found, False otherwise
+        :rtype: bool
+        """
+        if 'PATH' not in env:
+            raise KiwiCommandError('No PATH variable in environment')
+        return any(
+            os.access(os.path.join(path, command), os.X_OK)
+            for path in env['PATH'].split(os.pathsep)
+        )
```

Comment:
I did not see that which implementation, however I am not sure to use that Path.which method here, at least as it is now. I think the environment management is done in a different way in command than how it is done in  Path.which. I am not sure how the custom env you can give to a Command.run is actually used, but the command is not checked against the host environment if a custom one is provided. However, the Path.which, always includes the os.environ in the search. Is that the case or I am missing something?

If so I would suggest to slightly modify Path.which, even I am not entirely sure how to do that without altering the actual behavior of which and without adding non obvious extra parameters